src/commands/hosting-site-delete.ts

```diff
@@ -1,23 +1,51 @@
-import { bold } from "cli-color";
+import { bold, underline } from "cli-color";
 
 import { Command } from "../command";
-import { deleteSite } from "../hosting/api";
+import { logLabeledSuccess } from "../utils";
+import { getSite, deleteSite } from "../hosting/api";
+import { promptOnce } from "../prompt";
 import { FirebaseError } from "../error";
 import { requirePermissions } from "../requirePermissions";
 import * as getProjectId from "../getProjectId";
-import * as logger from "../logger";
 import * as requireConfig from "../requireConfig";
 
+const LOG_TAG = "hosting:site";
+
 export default new Command("hosting:site:delete <siteName>")
   .description("delete a Firebase Hosting site")
+  .option("-f, --force", "delete without confirmation")
   .before(requireConfig)
   .before(requirePermissions, ["firebasehosting.sites.delete"])
-  .action(async (siteName: string, options) => {
-    const projectId = getProjectId(options);
-    if (!siteName) {
-      throw new FirebaseError("siteName is required");
-    }
+  .action(
+    async (
+      siteName: string,
+      options: any // eslint-disable-line @typescript-eslint/no-explicit-any
+    ): Promise<void> => {
+      const projectId = getProjectId(options);
+      if (!siteName) {
+        throw new FirebaseError("siteName is required");
+      }
 
-    await deleteSite(projectId, siteName);
-    logger.info(`Site ${bold(siteName)} deleted.`);
-  });
+      let confirmed = Boolean(options.force);
+      if (!confirmed) {
+        confirmed = await promptOnce({
+          message: `Are you sure you want to delete the Hosting Site ${underline(
+            siteName
+          )} for project ${underline(projectId)}?`,
+          type: "confirm",
+          default: false,
+        });
+      }
+      if (!confirmed) {
+        return;
+      }
+
+      // Check that the site exists first, to avoid giving a sucessesful message on a non-existant site.
+      await getSite(projectId, siteName);
+      await deleteSite(projectId, siteName);
```

Comment:
IMO, the better pattern would be to catch any error from `deleteSite` and don't re-throw it if it's a 404 error. That saves a network call (and this pattern would error if the site didn't already exist, which I think is incorrect).

Reply:
WIth this, the getSite error message would say "Error: could not find site "site" for project "project""

The issue here is that, due to declarative apis, delete returns OK if the site doesn't exist or has already been deleted.  So the message ends up being that we've deleted a site that never existed on your project.   

If we want to stay closer to the rest API, I'd be OK with that behaviour, but would change the error message to say ` Successfully deleted site <site> for project <project>, or the site doesn't exist` (or something like that)

Reply:
Ugh, I know this conversation has happened before somewhere and I don't recall where...

Looking at `ext:uninstall`, it fails if a non-existent ID is provided, so I guess I'll bend on that and leaving this as-is sounds fine.